src/gui/editorwidgets/qgstexteditwrapper.cpp

```diff
@@ -74,6 +79,26 @@ QVariant QgsTextEditWrapper::value() const
     // input rather then discarding it entirely
     return QVariant( v.left( field().length() ) );
   }
+  else if ( field().type() == QVariant::Map )
+  {
+    if ( json::accept( v.toUtf8() ) )
+    {
+      QVariant qjson = QgsJsonUtils::parseJson( v.toStdString() );
+      mInvalidJSON = false;
+      return qjson;
+    }
+    else
+      // return null value if json is invalid
+    {
+      if ( field().length() > 0 )
```

Comment:
```suggestion
      if ( v.length() > 0 )
```
This should check the length of the value and not of the field. Or am I too confused to understand?